homeassistant/components/media_player/yamaha.py

```diff
@@ -0,0 +1,97 @@
+"""
+Support for Yamaha Receivers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.yamaha/
+"""
+import logging
+
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    MediaPlayerDevice)
+from homeassistant.const import STATE_OFF, STATE_ON
+REQUIREMENTS = ['rxv==0.1.9']
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_YAMAHA = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Yamaha platform."""
+    yamaha = YamahaDevice(config.get("name", "Yamaha Receiver"))
+    yamaha.update()
+    add_devices([yamaha])
+
+
+class YamahaDevice(MediaPlayerDevice):
+    """Representation of a Yamaha device."""
+
+    # pylint: disable=too-many-public-methods, abstract-method
+    def __init__(self, name):
+        """Initialize the Yamaha Receiver."""
+        import rxv
+        try:
+            self._receiver = rxv.find()[0]
+        # pylint:disable=broad-except
+        except Exception as exception:
+            _LOGGER.error("Error locating receiver. " + str(exception))
+            return
```

Comment:
You should catch and return in the `setup_platform` method. Otherwise you just cancel the constructor but the object will still get instantiated and added to Home Assistant - without having a receiver.